src/main/java/org/jenkinsci/maven/plugins/hpi/TestDependencyMojo.java

```diff
@@ -25,12 +26,14 @@
 public class TestDependencyMojo extends AbstractHpiMojo {
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
-        File testDir = new File(project.getBuild().getTestOutputDirectory(),"test-dependencies");
-        testDir.mkdirs();
-
+        File testDir = new File(project.getBuild().getTestOutputDirectory(), "test-dependencies");
         try {
-            Writer w = new OutputStreamWriter(new FileOutputStream(new File(testDir,"index")), StandardCharsets.UTF_8);
+            Files.createDirectories(testDir.toPath());
+        } catch (IOException e) {
+            throw new MojoExecutionException("Failed to create directories for '" + testDir + "'", e);
+        }
 
+        try (Writer w = new OutputStreamWriter(new FileOutputStream(new File(testDir, "index")), StandardCharsets.UTF_8)) {
```

Comment:
```suggestion
        try (FileOutputStream fos = new FileOutputStream(new File(testDir, "index")); Writer w = new OutputStreamWriter(fos, StandardCharsets.UTF_8)) {
```

Reply:
I suppose this is just for readability? I don't think there is a difference in behavior here. I have accepted the suggestion but it would be nice to understand the reasoning behind the suggestion.

Reply:
The former form reliably calls `OutputStreamWriter.close`, which will in turn call `FileOutputStream.close`, but only if it got as far as the `OutputStreamWriter` constructor. If the `FileOutputStream` is created but the `OutputStreamWriter` constructor fails, `FileOutputStream.close` will not be closed. That scenario is admittedly very unlikely in this case (I doubt the `OutputStreamWriter` constructor does anything risky), but at any rate the latter form ensures that both `close` calls are made. (IIUC)